test/test.ts

```diff
@@ -0,0 +1,19 @@
+import * as vhost from "../index"
+import * as connect from "connect"
+
+const app = connect()
+
+// Using application
+vhost("*.example.com", app)
+
+// Using request handler
+vhost("*.example.com",  (req, res, next) => {
+  if (req.vhost instanceof Array) {
```

Comment:
Is this ever true? I don't believe req.vhost is ever an array, right?

Reply:
I based this on the README (https://github.com/expressjs/vhost/blame/master/README.md#L48)

Reply:
There is no instanceof check in the readme.

Reply:
Got your point, my bad that I was only thinking in typings. I will change to the following and the problem should be solved.

```
  interface IVHost {
    [key: number]: string
    host: string,
    hostname: string,
    length: number
  }
```

Thanks!